scIB/integration.py

```diff
@@ -192,7 +192,88 @@ def runScvi(adata, batch, hvg=None):
     adata.obsm['X_emb'] = latent
 
     return adata
+def runScanvi(adata, batch, labels, hvg=None):
+# Use non-normalized (count) data for scanvi!
+    
+    # Check for counts data layer
+    if 'counts' not in adata.layers:
+        raise TypeError('Adata does not contain a `counts` layer in `adata.layers[`counts`]`')
+
+    from scvi.models import VAE, SCANVI
+    from scvi.inference import UnsupervisedTrainer, SemiSupervisedTrainer
+    from sklearn.preprocessing import LabelEncoder
+    from scvi.dataset import AnnDatasetFromAnnData
+    import numpy as np
+    
+    # STEP 1: prepare the data
+    net_adata = adata.copy()
+    net_adata.X = adata.layers['counts']
+    del net_adata.layers['counts']
+    # Ensure that the raw counts are not accidentally used
+    del net_adata.raw # Note that this only works from anndata 0.7
+
+    # Define batch indices
+    le = LabelEncoder()
+    net_adata.obs['batch_indices'] = le.fit_transform(net_adata.obs[batch].values)
+    net_adata.obs['labels'] = le.fit_transform(net_adata.obs[labels].values)
+
+    net_adata = AnnDatasetFromAnnData(net_adata)
+
+    print("scANVI dataset object with {} batches and {} cell types".format(net_adata.n_batches, net_adata.n_labels))
+
+    #if hvg is True:
+    #    # this also corrects for different batches by default
+    #    net_adata.subsample_genes(2000, mode="seurat_v3")
+
+    # # Defaults from SCVI github tutorials scanpy_pbmc3k and harmonization
+    n_epochs_scVI = np.min([round((20000/adata.n_obs)*400), 400]) #400
+    n_epochs_scANVI = np.min([10, np.max([2, round(n_epochs_scVI / 3.)])])
```

Comment:
`n_epochs_scANVI` seems very low as a default. This is definitely okay @romain-lopez?

Reply:
That's the value he proposed here: https://github.com/theislab/scib/pull/132#issuecomment-668870534

Reply:
okay, let's go with it!